lib/ts/BaseLogFile.h

```diff
@@ -55,16 +53,24 @@ typedef enum {
   LL_Fatal,     // causes process termination
 } LogLogPriorityLevel;
 
-#define log_log_trace(...)                         \
-  do {                                             \
-    if (BASELOGFILE_DEBUG_MODE)                    \
-      BaseLogFile::log_log(LL_Debug, __VA_ARGS__); \
+// log_log_* macros are turned on/off via the BASELOGFILE_DEBUG environment variable.
+// For example, run traffic_server like this: `BASELOGFILE_DEBUG=1 ./traffic_server`
+#define log_log_trace(...)                             \
+  do {                                                 \
+    static bool blf_debug_on = false;                  \
```

Comment:
Hmm, it looks like you are trying to avoid calling `getenv` if you can? But the common case is that this is not enabled and it call `getenv` every time on those cases. Perhaps:

```C
do {
  static enabled = getenv("FOO"); 
  if (enabled) {
       BaseLogFile::log_log(LL_Error, __VA_ARGS__);
  }
} while (0);
```